include/soci/soci-backend.h

```diff
@@ -294,6 +294,8 @@ class statement_backend
     virtual vector_into_type_backend* make_vector_into_type_backend() = 0;
     virtual vector_use_type_backend* make_vector_use_type_backend() = 0;
 
+    virtual db_type exchange_dbtype_for(db_type type) const { return type; }
```

Comment:
I know there are few comments in SOCI, but I'd like to change this and have more of them, at least for the new functions. How about something like this?

```suggestion
    // By default we suppose that there is a one-to-one mapping between DB
    // types and the types used in the database, but this may not be the case
    // for some backends (currently only SQLite3, see comment in its code), so
    // provide a way to use different intermediate types when using dynamic
    // (i.e. row-based) API.
    virtual db_type exchange_dbtype_for(db_type type) const { return type; }
```

Reply:
I like the premise. I decided to use a slightly different wording though.